contributor-guide/modules/ROOT/pages/version-control.adoc

```diff
@@ -1,4 +1,57 @@
-= Version Control (Git)
+= Version Control
 :navtitle: Version Control
 
-*Content under construction.*
+Boost uses the Git version control system for its development. This allows multiple contributors to work on the libraries concurrently, while keeping track of the changes that each person makes. Git is a distributed version control system, which means each contributor has their own local copy of the entire project, including its history.
+
+== Library Repositories
+
+The Boost libraries are modular, and each library has its own separate repository.  This makes it easier to work with the entire collection of Boost libraries.
+
+The official Boost repositories are hosted on GitHub at https://github.com/boostorg/[Boost.org]. Each library has its own repository under this organization. For example, https://github.com/boostorg/json for boost:json[], or https://github.com/boostorg/date_time/ for boost:date_time[].
+
+As with any Git-based project, you can clone the repositories, make changes, commit those changes to your local repository, and push your changes back to the server. If you want to contribute changes to the official Boost libraries, you can make a pull request, which the maintainers of the libraries can then review and potentially merge into the official codebase. Be aware though that Boost has a well-defined workflow for contributors, including guidelines for how to submit changes, how changes are reviewed, and so forth. Refer to xref:getting-involved.adoc[].
+
+== The Super-project Repository
+
+The Boost https://github.com/boostorg/boost[*Super-project*] repository includes all the libraries as submodules.
+
+In addition to the libraries, you'll find many meta-repositories under `boostorg` that handle other tasks, for example:
+
+* https://github.com/boostorg/website2022[Website generation]
+* https://github.com/boostorg/cmake[Cmake support]
+* https://github.com/boostorg/regression[Regression testing]
+* https://github.com/boostorg/release-tools[Release tools]
+
+There is also a https://github.com/boostorg/boost/tree/master/tools[tools] directory in the Super-project, which contains various tools used for tasks such as building and testing Boost libraries. 
+
+The *master* branch of the Super-project is usually the development version of Boost, where new features are added and bugs are fixed. When there is a public release of Boost, it is built from the Super-project repo.
```

Comment:
I'd suggest to rework & extend this a little bit.

* Each individual repository has a development (`develop`) branch, and a "this will go to prod the next release" branch (`master`). I'd say all the repos follow this, but you'd have to confirm it in Slack.
* The superproject does have these two branches: develop and master, with the same semantics.
* (git background) In git, when you add a submodule to a project, you make it point to a particular commit. This means that, when devs update sub-projects, the superproject doesn't get immediately updated.
* For this reason, there is a commit bot that runs nightly and updates the commit to which each submodule in the superproject points to. IIRC, this happens nightly and for both the `develop` and `master` branches. Please check the details on Slack too, since I do not maintain this.
* What are the implications of merging something in your library/submodule to develop? Due to how Boost is tested, you may affect other devs. As part of Boost testing, CIs usually clone the superproject and any required submodules. When testing any branch that is not master, the `develop` branch of the superproject is cloned. When you're testing `master` or a release, the `master` branch of the superproject is cloned. While this is not a requirement, most libs do it.
* Following the argument above, merging something to `develop` may affect any dependent code in the rest of Boost as soon as the commit bot makes the update to the superproject. E.g. if I do something wrong to MySQL develop branch, I will affect the `cmake` module, which runs tests to verify that you can use MySQL via cmake.
* In addition to what's listed above, the `master` branch dictates what will be in the next release. This is the branch that is frozen during the release process.
* It may be noteworthy that you, as a dev, don't directly create tags under repositories, ever. This differs from the usual workflow in other projects. Tags are created by the release scripts, both in the superproject and the individual repos.
* If you need to look at the code for the `1.82.0` release, just `git checkout boost-1.82.0` in the relevant repo.

Reply:
@anarthal - good info